### PR TITLE
Add running and failed status tab for DAGs on the UI

### LINKED ISSUE #17826: Add another tab in Dags table that shows running now
The DAGs main page has All/Active/Paused.

It would be nice if we can have also Running Now which will show all the DAGs that have at least 1 in Running of DAG runs:
Screen Shot 2021-08-25 at 11 30 55

For the above example the new DAGs tab of Running Now should show only 2 rows.
The use case is to see easily what DAGs are currently in progress.

### LINKED ISSUE #30280: Feature request - filter for dags with running status in the main page
### Description

Feature request to filter by running dags (or by other statuses too). We have over 100 dags and we were having some performance problems. we wanted to see all the running Dags from the main page and found that we couldn't. We can see the light green circle in the runs (and that involves a lot of scrolling) but no way to filter for it. 

We use SQL Server and it's job scheduling tool (SQL Agent) has this feature. The implementation for airflow shouldn't necessarily be like this but just presenting this as an example that it's a helpful feature implemented in other tools. 
image

I'll leave implementation details for someone else. 

on v2.2.5

### Use case/motivation

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by PApostol**
Add a "Running" tab on Airflow main page to filter for DAGs with `running` status.
Closes: #30280
Closes: https://github.com/apache/airflow/issues/17826

all_dags
running_dags

### CODE REVIEW

**Comment by PApostol** [comment]
A few tests are currently failing, e.g. in `api/experimental/test_endpoints.py`, `views/test_views_log.py`, etc, but I thought I'd raise a draft PR first to get some insights and/or pointers before I try to address them.

**Comment by dejlek** [comment]
Put it anywhere you like, this feature is very much needed, especially when you have hundreds of DAGs in your Airflow... I do not mind even if it goes to the menu somewhere, as long as I can run it whenever I want to see DAGs that have some tasks in running state...

> I second Elad's comment. Although we should make "Last Run" sortable and that should somewhat handle the use case.

Could we at least put it somewhere else, if you do not like it as tab near [All][Active][Paused] ? This is a feature I am waiting for too long...

**Comment by eladkal** [comment]
> @bbovenzi , could we at least put it somewhere else, if you do not like it as tab near [All][Active][Paused] ? This is a feature I am waiting for too long...

We could and we will but to complete this it requires involvement from the author of the PR to suggest another approch.
Please keep in mind that there are many types of users with different level of expertise. We must consider all of them and do our best to avoid causing confusion and problems.

**Comment by potiuk** [comment]
> Put it anywhere you like, this feature is very much needed, especially when you have hundreds of DAGs in your Airflow... I do not mind even if it goes to the menu somewhere, as long as I can run it whenever I want to see DAGs that have some tasks in running state...

> @bbovenzi , could we at least put it somewhere else, if you do not like it as tab near [All][Active][Paused] ? This is a feature I am waiting for too long...

I think constructive proposals and taking part in the discussion stating what is the best idea, and maybe even helping with mocking how it might look like @dejlek is far better way to speed it up than complaining? Another option is to create a PR with the change yourself if you are tired of waiting for something. Both approaches - trying to work out a good solution and active working on adding a feature are good, constructive ways of moving things forward.

Maybe you have a good proposal @dejlek where it could be added? If you have it and it fulfils the concerns raised above that would certainly speed up merging of the PR.

**Comment by tquazi** [comment]
To avoid confusion, can we move it away from the existing tab and create a new tab to filter based on the status of jobs. This will make the header a little busy.  
I think this can be a very useful feature to have when searching for Dags that are running. 

image

**Comment by bbovenzi** [comment]
> To avoid confusion, can we move it away from the existing tab and create a new tab to filter based on the status of jobs. This will make the header a little busy. I think this can be a very useful feature to have when searching for Dags that are running.
> 
> image

Good idea. I'd say go for it.

A few things to keep in mind:
- Add tooltips to describe what the tabs mean
- Make sure the css wrapping works for smaller window sizes

**Comment by eladkal** [comment]
> To avoid confusion, can we move it away from the existing tab and create a new tab to filter based on the status of jobs. This will make the header a little busy.
> I think this can be a very useful feature to have when searching for Da

Good idea!

**Comment by PApostol** [comment]
> To avoid confusion, can we move it away from the existing tab and create a new tab to filter based on the status of jobs. This will make the header a little busy. I think this can be a very useful feature to have when searching for Dags that are running.
> 
I like this idea, I'll give it a go and update the PR!

**Comment by PApostol** [comment]
I have tried to update the PR based on the suggestions and now the UI looks like below.  I'm not sure if that's the best way to do it (probably not), but my front-end skills are practically non-existent 😄. Happy for any suggestions and/or assistance!

running

**Comment by bbovenzi** [comment]
Looking good. Just a few points:

- Do you want to add a `failed` tab too?
- Let's update the tests at `tests_views_home.py`
- Bootstrap grid stuff, I think this will work:
    - All/Active/Paused: `col-md-3 col-sm-6`
    - Running/Failed: `col-md-3 col-sm-3`, we may need to center this content
    - Filter DAGs by tag: `col-md-2 col-sm-6`
    - Search DAGs: `col-md-2 col-sm-6`
    - Auto-refresh: `col-md-2 col-sm-3`, we may need to justify this content to the right

**Comment by bbovenzi** [comment]
> Thanks for the suggestions @bbovenzi, I'll give it a go. Regarding the `failed` tab, I was thinking about adding that, do you reckon it should be in this PR or in a separate one (linked with a separate feature request)?

Let's do it all together. It looks funny with only a "running" button

File: airflow/www/templates/airflow/dags.html [hunk 1]
```diff
@@ -132,22 +132,38 @@ <h2>{{ page_title }}</h2>
...
+          <div class="col-sm-3 col-md-3 text-center">
+            <div class="form-group btn-group">
+              <a
+                href="{{ url_for('Airflow.index', status='running', search=request.args.get('search', None), tags=request.args.getlist('tags', None)) }}"
+                class="btn {{ 'btn-primary' if status_filter == 'running' else 'btn-default' }}"
+                title="Show currently running DAG runs">Running <span class="badge">{{ "{:,}".format(status_count_running) }}</span></a>
+              <a
+                href="{{ url_for('Airflow.index', status='failed', search=request.args.get('search', None), tags=request.args.getlist('tags', None)) }}"
+                class="btn {{ 'btn-primary' if status_filter == 'failed' else 'btn-default' }}"
+                title="Show DAGs with failed latest DAG run">Failed <span class="badge">{{ "{:,}".format(status_count_failed) }}</span></a>
```
Comment (bbovenzi) [inline airflow/www/templates/airflow/dags.html]: The filter actually shows dags with any failed runs, not just the latest run.

**Comment by bbovenzi** [comment]
> I have included the other suggestions as well, however for All/Active/Paused grid I had to use `col-sm-6 col-md-6` instead of the suggested `col-sm-6 col-md-3` because the Running tab would overlap with the Paused tab (tested on 3440x1440, 2560x1440 and 1920x1080 screens).

Actually, I don't think bootstrap grid works well here.It would be better to switch this to a flex and wrap. But I can tackle that if you want help.

File: airflow/www/templates/airflow/dags.html (hunk 1 above)
Comment (PApostol) [reply airflow/www/templates/airflow/dags.html]: The join `subq_join` added in `views.py` actually ensures that only DAGs whose latest run is `FAILED` will appear in the Failed tab. This is so DAGs won't be stuck in the Failed tab if they have an old failed run. Instead, users would be shown only the DAGs that are currently failing and require action/mitigation. If a failed DAG is re-run and succeeds, it doesn't show in the Failed tab anymore. I feel if we show DAGs with any failed past runs, then there is not much useful information for users.

**Comment by PApostol** [comment]
> Actually, I don't think bootstrap grid works well here.It would be better to switch this to a flex and wrap. But I can tackle that if you want help.

Yes please, I've never touched CSS before I'm afraid!

**Comment by bbovenzi** [comment]
> > Actually, I don't think bootstrap grid works well here.It would be better to switch this to a flex and wrap. But I can tackle that if you want help.
> 
> Yes please, I've never touched CSS before I'm afraid!

Updated! Pull it down and let me know if you think that works.

**Review by bbovenzi — APPROVED** [review]

**Comment by PApostol** [comment]
> Updated! Pull it down and let me know if you think that works.

Alright, I think that works neatly. I noticed the "filter" and "search" boxes have become a bit smaller, if that's of any concern. This is how it currently looks like (2560x1440 screen, Firefox, Ubuntu):

ui

**Pull request merged**